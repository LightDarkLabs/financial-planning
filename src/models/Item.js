import { types } from 'mobx-state-tree'
import Money from '@models/Money'

const Item = types
  .model({
    date: types.Date,
    value: types.optional(Money, {})
  })

export default Item
