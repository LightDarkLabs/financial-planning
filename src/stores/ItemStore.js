import { types } from 'mobx-state-tree'
import Item from '@models/Item'

const ItemStore = types
  .model({
    items: types.optional(types.array(Item), [])
  })
  .named('ItemStore')

export default ItemStore
