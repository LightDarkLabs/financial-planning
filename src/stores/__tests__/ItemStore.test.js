import { getSnapshot } from 'mobx-state-tree'
import Item from '@models/Item'
import { stubDate } from '@testhelper'
import ItemStore from '@stores/ItemStore'

stubDate(new Date('2018-01-01T00:00:00Z'))

it('serializes', () => {
  const store = ItemStore.create({})

  expect(getSnapshot(store)).toMatchSnapshot()
})
