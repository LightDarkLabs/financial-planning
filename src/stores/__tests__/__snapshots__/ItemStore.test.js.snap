// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serializes 1`] = `
Object {
  "items": Array [],
}
`;
